0.4.0 revised documentation (#428)
mehta-lab/recOrder: docs/napari-plugin-guide.md

```
napari -w recOrder-napari
```

> ## Connect to MicroManager

Reviewer: Not a big deal, but the official spelling seems to be [Micro-Manager](https://micro-manager.org/Micro-Manager_User's_Guide).

Author: Fixed. Thanks!

Click “Connect to MM”. If the connection succeeds, proceed to calibration. If not, revisit the [microscope installation guide](./microscope-installation-guide.md).

![](./images/connect_to_mm.png)

## Calibration tab

> The first step in the acquisition process is to calibrate the liquid crystals and measure a background. In the `recOrder` plugin you will see the following options for controlling the calibration:

Reviewer: Should we note that this is only necessary for polarization imaging?

Author: Good call, fixed.

![](./images/run_calib.png)

### Prepare for a calibration

> Place your sample on the stage, focus on the surface of the coverlip/well, navigate to **an empty FOV**, then align the light source into **Kohler illumination** [following these steps.](https://www.microscopyu.com/tutorials/kohler)

Reviewer: ```suggestion
Place your sample on the stage, focus on the surface of the coverslip/well, navigate to **an empty FOV**, then align the light source into **Kohler illumination** [following these steps](https://www.microscopyu.com/tutorials/kohler).
```

Choose an **Illumination Scheme** to decides how many polarization states you will calibrate and use. We recommend starting with the *4-State (Ext, 0, 60, 120)* scheme as it requires one less illumination state than the *5-State* scheme.

**Calibration Mode** is set automatically, so the default value is a good place to start. Different modes allow calibrations with voltages, retardances, or hardware sequencing. 

The **Config Group** is set automatically to the MicroManager configuration group that contains the `State*` presets. You can modify this option if you have multple configuration groups with these presets. 

### Run the calibration
Start a calibration with **Run Calibration**. 

The progress bar will show the progress of calibration, and it should take less than 2 minutes on most systems.

The plot shows the intensities over time during calibration. One way to diagnose an in-progress calibration is to watch the intensity plot. An ideal plot will look similar to the following:

![](./images/ideal_plot.png)

> Once finished, you will get a calibration assessment and an extinction value. The extinction value  gives you a metric for calibration quality: the higher the extinction, the cleaner the light path and the greater the sensitivity of QLIPP.

Reviewer: ```suggestion
Once finished, you will get a calibration assessment and an extinction value. The extinction value gives you a metric for calibration quality: the higher the extinction, the cleaner the light path and the greater the sensitivity of QLIPP.
```

* **Extinction 0 – 50**:  Very poor. The alignment of the universal compensator may be off or the sample chamber may be highly birefringent. 

* **Extinction 50 - 100**: Okay extinction, could be okay for tissue imaging and strong anisotropic structures. Most likely not suitable for cell imaging

* **Extinction 100 - 200**: Good Extinction. These are the typical values we get on our microscopes.

> * **Extinction 200**+: Excellent. Indicates a very well-aligned and clean light path and high sensitivity of the system.

Reviewer: ```suggestion
* **Extinction 200+**: Excellent. Indicates a very well-aligned and clean light path and high sensitivity of the system.
```

### Acquire Buttons
![](./images/acquire_buttons.png)

The **Retardance + Orientation**, **Phase From BF**, and **Retardance + Orientation + Phase** buttons set off MicroManager acquisitions that use the upcoming acquisition settings. After the acquisition is complete, these routines will set off `recOrder` reconstructions that estimate the named parameters. 

The **STOP** button will end the acquisition as soon as possible, though MicroManager acquisitions cannot always be interrupted. 

### Acquisition Settings
![](./images/acquisition_settings.png)

The **Acquisition Mode** sets the target dimensions for the reconstruction. Perhaps surprisingly, all 2D reconstructions require 3D data except for **Retardance + Orientation** in **2D Acquisition Mode**. The following table summarizes the data that will be acquired when an acquisition button is pressed in **2D** and **3D** acquisition modes:

| **Acquisition** \ Acquisition Mode | 2D mode | 3D mode |  
| :--- | :--- | :--- |

> | **Retardance + Orientation** | CYX data | CYX data | 

Reviewer: The previous paragraph says:

> The following table summarizes the data that will be acquired

So in the case of 3D birefringence, although the reconstruction only sees a CYX at a time, CZYX will still be acquired (and reconstructed).

Author: That was an error! Good eye. Retardance + Orientation 3D should be CZYX.

Author: Fixed

| **Phase From BF** | ZYX data | ZYX data | 
| **Retardance + Orientation + Phase** | CZYX data | CZYX data | 

> Unless a **Retardance + Orientation** reconstruction in **2D Acquisition Mode** is requested, `recOrder` uses MicroManager's z-stage to acquire 3D data. **Z Start**, **Z End**, and **Z Step** are stage settings for acquiring an image volume, relative to the current position of the stage. Values are in the stage's default units, typically in microns.

Reviewer: ```suggestion
Unless a **Retardance + Orientation** reconstruction in **2D Acquisition Mode** is requested, `recOrder` uses MicroManager's z-stage to acquire 3D data. **Z Start**, **Z End**, and **Z Step** are stage settings for acquiring an image volume, relative to the current position of the stage. Values are in the stage's default units, typically in micrometers.
```

Author: Thanks, changed all "microns" to "micrometers".

The **Background Correction** menu has several options (each with mouseover explanations):
* **None**: No background correction is performed. 
* **Measured**: Corrects sample images with a background image acquired at an empty field of view, loaded from **Background Path**, by default the most recent background acquisition. 
* **Estimated**: Estimates the sample background by fitting a 2D surface to the sample images. Works well when structures are spatially distributed across the field of view and a clear background is unavailable.
* **Measured + Estimated**: Applies a **Measured** background correction then an **Estimated** background correction. Use to remove residual background after the sample retardance is corrected with measured background.

The remaining parameters are used by the reconstructions:
 
* **GPU ID**: Not implemented
* **Wavelength (nm)**: illumination wavelength
* **Objective NA**: numerical aperture of the objective, typically found next to magnification
* **Condenser NA**: numerical aperture of the condenser

> * **Camera Pixel Size (um)**: pixel size of the camera in microns (e.g. 6.5 um)

Reviewer: ```suggestion
* **Camera Pixel Size (um)**: pixel size of the camera in micrometers (e.g. 6.5 μm)
```

* **RI of Obj. Media**: refractive index of the objective media, typical values are 1.0 (air), 1.3 (water), 1.473 (glycerol), or 1.512 (oil)
* **Magnification**: magnification of the objective
* **Rotate Orientation (90 deg)**: rotates "Orientation" reconstructions by +90 degrees clockwise and saves the result, most useful when a known-orientation sample is available

> * **Flip Orientation (90 deg)**: flips "Orientation" reconstructions about napari's horizontal axis and saves the result

Reviewer: ```suggestion
* **Flip Orientation**: flips "Orientation" reconstructions about napari's horizontal axis before saving the result
```

* **Invert Phase Contrast**: inverts the phase reconstruction's contrast by flipping the positive and negative directions of the stage during the reconstruction, and saves the result 

### Phase Reconstruction Settings
![](./images/phase_reconstruction_settings.png)

These parameters are used only by phase reconstructions

> * **Z Padding**: The number of slices to pad on either end of the stack in order to correct for edge reflection artifacts, necessary if the sample is not fully out of focus on either end of the stack

Reviewer: Is this reflection padding?

Author: Not quite. Simplified.
